hackernews/tests/test_api.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+import logging
+from provider import UpstreamProviderError
+
+
+def test_search_success(authed_client, mock_search_provider):
+    response = authed_client.post(
+        "/search",
+        json={"query": "test"},
+    )
+
+    assert response.status_code == 200
+    # Mock returns [] for search results
+    assert response.get_json() == {"results": []}
+
+
+def test_search_query_is_missing_fail(authed_client):
+    response = authed_client.post(
+        "/search",
+        json={},
+    )
+
+    assert response.get_json() == {
+        "detail": "'query' is a required property",
+        "status": 400,
+        "title": "Bad Request",
+        "type": "about:blank",
+    }
+
+
+def test_search_query_is_empty_string_fail(authed_client):
+    response = authed_client.post(
+        "/search",
+        json={"query": ""},
+    )
+
+    assert response.get_json() == {
+        "detail": "'' is too short - 'query'",
+        "status": 400,
+        "title": "Bad Request",
+        "type": "about:blank",
+    }
```

Comment:
Yep, connexion and the openapi definition - I still would have preferred some testing around request handling without going too deep (since `connexion` itself is already well-tested). Do you prefer removing these, or just keeping it as-is without adding more?